test/ui/signin/sign_in_view_model_test.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:survey_flutter_ic/api/exception/network_exceptions.dart';
+import 'package:survey_flutter_ic/ui/signin/sign_in_view_model.dart';
+import 'package:survey_flutter_ic/ui/signin/sign_in_view_state.dart';
+import 'package:survey_flutter_ic/usecase/base/base_use_case.dart';
+
+import '../../mocks/generate_mocks.mocks.dart';
+
+void main() {
+  group('SignInViewModel', () {
+    late MockSignInUseCase mockSignInUseCase;
+    late ProviderContainer container;
+
+    setUp(() {
+      mockSignInUseCase = MockSignInUseCase();
+
+      container = ProviderContainer(overrides: [
+        signInViewModelProvider
+            .overrideWith((ref) => SignInViewModel(mockSignInUseCase))
+      ]);
+      addTearDown(() => container.dispose());
```

Comment:
Let's create `late SignInViewModel signInViewModel;`
```suggestion
      addTearDown(() => container.dispose());
      signInViewModel = container.read(signInViewModelProvider.notifier);
```

Reply:
👍 Makes sense